### PR TITLE
Chore: (Docs) Updates to the Angular documentation

### PR DESCRIPTION
**Pull request by jonniebigodes**
With this pull request the documentation is updated to feature the Angular Storybook custom builder. Follows up on #17380 and #17305 .

What was done:
- Updated the get-started/Install section of the documentation and subsequent snippets to include a custom builder.
- Updated the Configure/Styling and CSS docs to mention the updated version of using CSS with Storybook's builder.
- Updated the Sharing/Publish documentation to feature the angular builder also.

@shilman and @storybookjs/angular would appreciate some feedback on this when you're able. Thanks in advance.

### CODE REVIEW

File: docs/snippets/angular/build-storybook-production-mode.with-builder.js.mdx [hunk 1]
```diff
@@ -0,0 +1,6 @@
+```shell
+# Builds Storybook with Angular's custom builder
+# See https://storybook.js.org/docs/angular/get-started/install 
+# to learn how to create the custom builder
+ng run my-project:build-storybook
```
Comment (shilman) [inline docs/snippets/angular/build-storybook-production-mode.with-builder.js.mdx:5]: we might be able to just instruct users to add this to their `"scripts"` tag and then the rest of the instructions could stay the same. WDYT?

```json
{
  "scripts": {
     "build-storybook": "ng run my-project:build-storybook",
  }
}
```

**Review by shilman — COMMENTED** [review]
Looking good! Misc comments, nothing serious

File: docs/get-started/installation-problems/angular.mdx [hunk 2]
```diff
@@ -12,6 +12,28 @@
 
   Check the [Migration Guide](https://github.com/storybookjs/storybook/blob/next/MIGRATION.md#automigrate) for more information on how to set up Webpack 5.
 
+- If you're using the custom Storybook builder, Storybook will not automatically open a browser window or tab. It's is a known issue that we plan to fix in future releases.
```
[the remark this answers is not in the document]
Comment (jonniebigodes) [reply docs/get-started/installation-problems/angular.mdx:15]: Don't mention it. Caught it during testing :)!

File: docs/configure/styling-and-css.md [hunk 3]
```diff
@@ -28,6 +28,37 @@ If your component files import their CSS, Storybook’s webpack config will work
 
 <!-- prettier-ignore-end -->
 
+- With Angular version 13 and above, you should use a builder configuration to import your CSS:
```
[the remark this answers is not in the document]
Comment (jonniebigodes) [reply docs/configure/styling-and-css.md]: Indeed you're correct. I've updated this section and used a similar approach as it's being used in the Get Started section with my latest commit.

File: docs/snippets/angular/build-storybook-production-mode.with-builder.js.mdx (hunk 1 above)
Comment (jonniebigodes) [reply docs/snippets/angular/build-storybook-production-mode.with-builder.js.mdx:5]: I've updated the Publish Storybook and factored it in with the latest commit.

**Review by shilman — APPROVED** [review]
Great stuff @jonniebigodes this is awesome 💯

**Comment by theiliad** [comment]
Hi @jonniebigodes @shilman 

Can we actually add a sample project here?

I've made a setup in angular 13, building fine, however storybook is not liking it at all...

I've tried all kinds of things for ~3hrs today. webpack 4, webpack 5, `npx sb prerelease`, looked at sample repos (e.g. angular cli, chroma, only they all seem to be using angular 11 or 12...)

**Comment by shilman** [comment]
@theiliad We'll follow up with example projects in the next couple months, but not as part of this PR.

**Comment by theiliad** [comment]
> @theiliad We'll follow up with example projects in the next couple months, but not as part of this PR.

That's fair. Just so I can make some progress though, could we share the repo/setup that was used to test with angular13?

I can just figure it out from there

**Comment by shilman** [comment]
@theiliad Unfortunately we don't have a repo for that. I was playing around with @kroeder last week to try to better understand where we were at with Angular support and created a repo then. I don't remember where we got to, but I'm sharing that here in case it helps: https://github.com/shilman/storybook-angular13

Most of the inspiration was taken from the monorepo, which is running an older version of Angular but which uses the new config format https://github.com/storybookjs/storybook/tree/next/examples/angular-cli

You can also see some migration docs that came out of that session here: https://github.com/storybookjs/storybook/pull/17380

Sorry it's a bit all over the place, but that's where we're at right now.

**Comment by theiliad** [comment]
> @theiliad Unfortunately we don't have a repo for that. I was playing around with @kroeder last week to try to better understand where we were at with Angular support and created a repo then. I don't remember where we got to, but I'm sharing that here in case it helps: https://github.com/shilman/storybook-angular13
> 
> Most of the inspiration was taken from the monorepo, which is running an older version of Angular but which uses the new config format https://github.com/storybookjs/storybook/tree/next/examples/angular-cli
> 
> You can also see some migration docs that came out of that session here: #17380
> 
> Sorry it's a bit all over the place, but that's where we're at right now.

Thanks! This is already helping me.

Haven't gotten it to fully work yet, but FYI, 1 of the errors that said nothing about compodoc was fixed by adding
```
    "@compodoc/compodoc": "^1.1.18"
```

to **package.json**

**Pull request merged**